packages/node/src/integrations/tracing/fastify/index.ts

```diff
@@ -0,0 +1,225 @@
+import { FastifyOtelInstrumentation } from '@fastify/otel';
+import type { FastifyInstance, FastifyReply, FastifyRequest } from './types';
+import {
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  captureException,
+  defineIntegration,
+  getClient,
+  getIsolationScope,
+  logger,
+  spanToJSON,
+} from '@sentry/core';
+import type { IntegrationFn, Span } from '@sentry/core';
+import { generateInstrumentOnce } from '../../../otel/instrument';
+import { FastifyInstrumentationV3 } from './v3/instrumentation';
+import * as diagnosticsChannel from 'node:diagnostics_channel';
+import { DEBUG_BUILD } from '../../../debug-build';
+
+interface FastifyHandlerOptions {
+  /**
+   * Callback method deciding whether error should be captured and sent to Sentry
+   *
+   * @param error Captured Fastify error
+   * @param request Fastify request (or any object containing at least method, routeOptions.url, and routerPath)
+   * @param reply Fastify reply (or any object containing at least statusCode)
+   *
+   * @example
+   *
+   * ```javascript
+   * setupFastifyErrorHandler(app, {
+   *   shouldHandleError(_error, _request, reply) {
+   *     return reply.statusCode >= 400;
+   *   },
+   * });
+   * ```
+   *
+   * If using TypeScript, you can cast the request and reply to get full type safety.
+   *
+   * ```typescript
+   * import type { FastifyRequest, FastifyReply } from 'fastify';
+   *
+   * setupFastifyErrorHandler(app, {
+   *   shouldHandleError(error, minimalRequest, minimalReply) {
+   *     const request = minimalRequest as FastifyRequest;
+   *     const reply = minimalReply as FastifyReply;
+   *     return reply.statusCode >= 500;
+   *   },
+   * });
+   * ```
+   */
+  shouldHandleError: (error: Error, request: FastifyRequest, reply: FastifyReply) => boolean;
+}
+
+const INTEGRATION_NAME = 'Fastify';
+const INTEGRATION_NAME_V3 = 'Fastify-V3';
+
+export const instrumentFastifyV3 = generateInstrumentOnce(INTEGRATION_NAME_V3, () => new FastifyInstrumentationV3());
+
+export const instrumentFastify = generateInstrumentOnce(INTEGRATION_NAME, () => {
+  const fastifyOtelInstrumentationInstance = new FastifyOtelInstrumentation();
+  const plugin = fastifyOtelInstrumentationInstance.plugin();
+
+  // This message handler works for Fastify versions 3, 4 and 5
```

Comment:
this also works for fastify 3? I thought only 4+5?

Reply:
Yes, only this channel works on fastify 3 - [v3 Docs](https://fastify.dev/docs/v3.29.x/Reference/Hooks/#diagnostics-channel-hooks)

This will. I think, make it possible to remove `setupFastifyErrorHandler` call everywhere.
I can add that in a separate PR, or also can add here, making `setupFastifyErrorHandler` a no-op until we deprecate it.